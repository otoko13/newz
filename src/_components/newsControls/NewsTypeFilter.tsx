import React from 'react';
import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
import { ENewsType } from '../newsService';

export interface INewsTypeFilterProps {
    filters: string[];
    onFiltersChanged: (types: string[]) => void;
}

const NewsTypeFilter = (props: INewsTypeFilterProps) => {

    function handleChange(event: React.FormEvent<HTMLDivElement>, item?: IDropdownOption) {
        if (!item) {
            return;
        }
        const newSelectedItems = [...props.filters];
        if (item && item.selected) {
          // add the option if it's checked
          newSelectedItems.push(item.key as string);
        } else {
          // remove the option if it's unchecked
          const currIndex = newSelectedItems.indexOf(item.key as string);
          if (currIndex > -1) {
            newSelectedItems.splice(currIndex, 1);
          }
        }
        props.onFiltersChanged(newSelectedItems);
    }

    return (
        <Dropdown
            className='NewsTypeFilter'
            placeholder="Select types of articles"
            label="What do you want to see?"
            selectedKeys={props.filters}
            onChange={handleChange}
            multiSelect
            options={[
                { key: ENewsType.Story.toString(), text: 'Stories' },
                { key: ENewsType.Job.toString(), text: 'Jobs' },
                { key: ENewsType.Poll.toString(), text: 'Polls' },
            ]}
            styles={{ dropdown: { width: 200 } }}
        />
    );
}

export default NewsTypeFilter;