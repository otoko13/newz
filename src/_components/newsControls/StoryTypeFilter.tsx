import React from 'react';
import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
import { EStoriesType } from '../newsService';

export interface IStoryTypeFilterProps {
    filter: string;
    onFilterChanged: (type: string) => void;
}

const StoryTypeFilter = (props: IStoryTypeFilterProps) => {

    function handleChange(_event: React.FormEvent<HTMLDivElement>, item?: IDropdownOption) {
        if (item) {
            props.onFilterChanged(item.key as string);
        }
    }

    return (
        <Dropdown
            className='StoryTypeFilter'
            selectedKey={props.filter}
            onChange={handleChange}
            options={[
                { key: EStoriesType.New.toString(), text: 'Newest' },
                { key: EStoriesType.Top.toString(), text: 'Top stories' },
                { key: EStoriesType.Best.toString(), text: 'Best stories' },
            ]}
            styles={{ dropdown: { width: 200 } }}
        />
    );
}

export default StoryTypeFilter;