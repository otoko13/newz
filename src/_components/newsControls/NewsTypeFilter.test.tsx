import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import StoryTypeFilter, { IStoryTypeFilterProps } from './StoryTypeFilter';
import NewsTypeFilter, { INewsTypeFilterProps } from './NewsTypeFilter';

let props: INewsTypeFilterProps;

beforeEach(() => {
    props = {
        filters: ['story'],
        onFiltersChanged: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<NewsTypeFilter {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('changing filters', () => {
    it('should call props.onFiltersChanged with the correct new filters', () => {
        const { getAllByRole, getAllByTitle } = render(<NewsTypeFilter {...props} />);
        fireEvent.click(getAllByRole('button')[0]);
        fireEvent.click(getAllByTitle('Stories')[1]);
        fireEvent.click(getAllByTitle('Polls')[1]);
        fireEvent.click(getAllByTitle('Jobs')[1]);
        expect(props.onFiltersChanged).toHaveBeenCalledWith(['story', 'job']);
    });
});