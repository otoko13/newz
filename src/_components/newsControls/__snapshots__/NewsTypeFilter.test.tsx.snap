// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ms-Dropdown-container NewsTypeFilter root-40"
>
  <label
    class="ms-Label ms-Dropdown-label label-58"
    id="Dropdown0-label"
  >
    What do you want to see?
  </label>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="Dropdown0-label Dropdown0-option"
    class="ms-Dropdown dropdown-42"
    data-is-focusable="true"
    id="Dropdown0"
    role="button"
    tabindex="0"
  >
    <span
      aria-atomic="true"
      aria-invalid="false"
      aria-live="polite"
      class="ms-Dropdown-title title-43"
      id="Dropdown0-option"
    >
      Stories
    </span>
    <span
      class="ms-Dropdown-caretDownWrapper caretDownWrapper-44"
    >
      <i
        aria-hidden="true"
        class="ms-Dropdown-caretDown caretDown-59"
        data-icon-name="ChevronDown"
      />
    </span>
  </div>
</div>
`;
